### PR TITLE
operator: migrate test framework to testify

### PR DESCRIPTION
**Pull request by lhy1024**
Signed-off-by: lhy1024 <admin@liudos.us>

### What problem does this PR solve?

Issue Number: Ref https://github.com/tikv/pd/issues/4813.

### Check List

Tests

- Unit test

### Release note

```release-note
None.
```

### CODE REVIEW

**Review by JmPotato — APPROVED** [review]

File: server/schedule/operator/builder_test.go [hunk 1]
```diff
@@ -541,71 +546,73 @@ func (s *testBuilderSuite) TestBuild(c *C) {
...
+func (suite *operatorBuilderTestSuite) TestTargetUnhealthyPeer() {
 	p := &metapb.Peer{Id: 2, StoreId: 2, Role: metapb.PeerRole_Learner}
 	region := core.NewRegionInfo(&metapb.Region{Id: 1, Peers: []*metapb.Peer{{Id: 1, StoreId: 1},
 		p}}, &metapb.Peer{Id: 1, StoreId: 1}, core.WithPendingPeers([]*metapb.Peer{p}))
-	builder := NewBuilder("test", s.cluster, region)
+	builder := NewBuilder("test", suite.cluster, region)
 	builder.PromoteLearner(2)
-	c.Assert(builder.err, NotNil)
+	suite.Error(builder.err)
+	suite.NotNil(builder.err)
```
Comment (rleungx) [inline server/schedule/operator/builder_test.go:598]: remove it?

File: server/schedule/operator/builder_test.go [hunk 2]
```diff
@@ -541,71 +546,73 @@ func (s *testBuilderSuite) TestBuild(c *C) {
...
 	p = &metapb.Peer{Id: 2, StoreId: 2, Role: metapb.PeerRole_Voter}
 	region = core.NewRegionInfo(&metapb.Region{Id: 1, Peers: []*metapb.Peer{{Id: 1, StoreId: 1},
 		p}}, &metapb.Peer{Id: 1, StoreId: 1}, core.WithPendingPeers([]*metapb.Peer{p}))
-	builder = NewBuilder("test", s.cluster, region)
+	builder = NewBuilder("test", suite.cluster, region)
 	builder.SetLeader(2)
-	c.Assert(builder.err, NotNil)
+	suite.Error(builder.err)
+	suite.NotNil(builder.err)
```
Comment (rleungx) [inline server/schedule/operator/builder_test.go:611]: ditto

File: server/schedule/operator/builder_test.go [hunk 3]
```diff
@@ -541,71 +546,73 @@ func (s *testBuilderSuite) TestBuild(c *C) {
...
 	region = core.NewRegionInfo(&metapb.Region{Id: 1, Peers: []*metapb.Peer{{Id: 1, StoreId: 1},
 		p}}, &metapb.Peer{Id: 1, StoreId: 1}, core.WithDownPeers([]*pdpb.PeerStats{{Peer: p}}))
-	builder = NewBuilder("test", s.cluster, region)
+	builder = NewBuilder("test", suite.cluster, region)
 	builder.SetLeader(2)
-	c.Assert(builder.err, NotNil)
+	suite.Error(builder.err)
+	suite.NotNil(builder.err)
```
Comment (rleungx) [inline server/schedule/operator/builder_test.go:617]: ditto

File: server/schedule/operator/builder_test.go [hunk 4]
```diff
@@ -541,71 +546,73 @@ func (s *testBuilderSuite) TestBuild(c *C) {
...
 			case AddLearner:
-				c.Assert(step.ToStore, Equals, tc.steps[i].(AddLearner).ToStore)
+				suite.Equal(tc.steps[i].(AddLearner).ToStore, step.ToStore)
 			case PromoteLearner:
-				c.Assert(step.ToStore, Equals, tc.steps[i].(PromoteLearner).ToStore)
+				suite.Equal(tc.steps[i].(PromoteLearner).ToStore, step.ToStore)
 			case ChangePeerV2Enter:
-				c.Assert(len(step.PromoteLearners), Equals, len(tc.steps[i].(ChangePeerV2Enter).PromoteLearners))
-				c.Assert(len(step.DemoteVoters), Equals, len(tc.steps[i].(ChangePeerV2Enter).DemoteVoters))
+				suite.Equal(len(tc.steps[i].(ChangePeerV2Enter).PromoteLearners), len(step.PromoteLearners))
```
Comment (rleungx) [inline server/schedule/operator/builder_test.go:569]:
```suggestion
				suite.Len(tc.steps[i].(ChangePeerV2Enter).PromoteLearners, len(step.PromoteLearners))
```

**Review by rleungx — APPROVED** [review]

**Pull request merged**